New Resource & Data Source: azurerm_data_migration_service (#5258)
terraform-providers/terraform-provider-azurerm: website/docs/r/data_migration_service.html.markdown

> ---

Reviewer: can we add this to the sidebar too?

Author: Do you mean we should not add it into sidebar for now?

Reviewer 2: It should be added to the website index erb so it shows up in the nav

subcategory: "Data Migration"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_data_migration_service"
sidebar_current: "docs-azurerm-resource-data-migration-service"
description: |-

>   Manage Azure Service instance.

Reviewer: ```suggestion
  Manages a Data Migration Service.
```

---

# azurerm_data_migration_service

> Manage Azure Data Migration Service instance.

Reviewer: ```suggestion
Manages a Data Migration Service.
```

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-rg"

>   location = "northeurope"

Reviewer: we use West Europe for all examples - as such could we make this:

```suggestion
  location = "West Europe"
```

  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "example" {
  name 				   = "example-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefix       = "10.0.1.0/24"
}

resource "azurerm_data_migration_service" "example" {
	name                = "example-dms"
	location            = azurerm_resource_group.example.location
	resource_group_name = azurerm_resource_group.example.name
	virtual_subnet_id   = azurerm_subnet.example.id
	sku_name            = "Standard_1vCores"
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specify the name of the data migration service. Changing this forces a new resource to be created.

* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) Name of the resource group in which to create the data migration service. Changing this forces a new resource to be created.

> * `virtual_subnet_id` - (Required) The ID of the virtual subnet resource to which the data migration service should be joined. Changing this forces a new resource to be created.

Reviewer: should we be making this `subnet_id` to match other resources?

Author: Yes, my mistake 😥

> * `sku_name` - (Required) The sku name of the data migration service. Currently only one of "Premium_4vCores", "Standard_1vCores", "Standard_2vCores", "Standard_4vCores" is allowed. Changing this forces a new resource to be created.

Reviewer: can we quote these to match other resources:

```suggestion
* `sku_name` - (Required) The sku name of the data migration service. Possible values are `Standard_1vCores`, `Standard_2vCores`, `Standard_4vCores` and `Premium_4vCores`. Changing this forces a new resource to be created.
```

> * `kind` - (Optional) The resource kind. Only 'Cloud' (the default) is supported. Changing this forces a new resource to be created.

Reviewer: (as above) we can remove this

Reviewer 2: same comment

* `tags` - (Optional) A mapping of tags to assigned to the resource.

## Attributes Reference

The following attributes are exported:

> * `type` - (Optional) The resource type chain (e.g. virtualMachines/extensions)

Reviewer: this field doesn't seem useful to export - can we remove it?

> * `id` - Resource ID of data migration service.

Reviewer: we don't use the term "Resource ID" since it's ambiguous between Terraform and Azure - could we instead make this:

```suggestion
* `id` - The ID of Data Migration Service.
```

to match the other resources

## Import

> Data Migration Service can be imported using the `resource id`, e.g.

Reviewer: ```suggestion
Data Migration Services can be imported using the `resource id`, e.g.
```

	sku_name            = "Standard_1vCores"
}
```

## Argument Reference

The following arguments are supported:

* `location` - (Required) The Azure region of the operation. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) Name of the resource group. Changing this forces a new resource to be created.

* `name` - (Required) Name of the service. Changing this forces a new resource to be created.

* `virtual_subnet_id` - (Required) The ID of the virtual subnet resource to which the service should be joined. Changing this forces a new resource to be created.

* `sku_name` - (Required) The sku name of the service. Changing this forces a new resource to be created.

* `kind` - (Optional) The resource kind. Only 'Cloud' (the default) is supported. Changing this forces a new resource to be created.

* `delete_running_tasks` - (Optional) When destroy the resource, delete it even if it contains running tasks.

* `tags` - (Optional) Resource tags.

## Attributes Reference

The following attributes are exported:

* `type` - (Optional) The resource type chain (e.g. virtualMachines/extensions)

* `provisioning_state` - The resource's provisioning state

> * `id` - Resource ID.

Reviewer 2: could we make this read like the rest of the resources? and be a be more clear what some of these properties do? ie type
